fdbclient/admin_client.go

```diff
@@ -529,6 +529,16 @@ func (client *cliAdminClient) ChangeCoordinators(addresses []fdbv1beta2.ProcessA
 
 // GetConnectionString fetches the latest connection string.
 func (client *cliAdminClient) GetConnectionString() (string, error) {
+	adminClientMutex.Lock()
+	defer adminClientMutex.Unlock()
+
+	version, err := fdbv1beta2.ParseFdbVersion(client.Cluster.Spec.Version)
+	// connectionString key is supported in 6.x
+	if client.useClientLibrary && err == nil && version.IsSupported() {
```

Comment:
I think in addition to that we want to have a feature gate to enable this new behaviour otherwise this will be the default behaviour.